tests/admin_widgets/tests.py

```diff
@@ -1294,3 +1294,210 @@ def test_ForeignKey_using_to_field(self):
         profiles = Profile.objects.all()
         self.assertEqual(len(profiles), 1)
         self.assertEqual(profiles[0].user.username, username_value)
+
+    def test_ForeignKey_as_Select(self):
+        self.admin_login(username='super', password='secret', login_url='/')
+        self.selenium.get(self.live_server_url + reverse('admin:admin_widgets_team_add'))
+
+        main_window = self.selenium.current_window_handle
+        # Only an empty option in the Select
+        self.assertEqual(len(self.selenium.find_elements_by_css_selector('#id_team_manager option')), 1)
+
+        # Click the Add Manager button to add new
+        self.selenium.find_element_by_id('add_id_team_manager').click()
+        self.wait_for_popup()
+        self.selenium.switch_to.window('id_team_manager')
+
+        manager_name_field = self.selenium.find_element_by_id('id_name')
+        manager_name_field.send_keys('Martin')
+
+        save_button_css_selector = '.submit-row > input[type=submit]'
+        self.selenium.find_element_by_css_selector(save_button_css_selector).click()
+        self.selenium.switch_to.window(main_window)
+
+        self.assertEqual(len(self.selenium.find_elements_by_css_selector('#id_team_manager option')), 2)
+        new_manager = Manager.objects.get(pk=1)
+        self.assertEqual(new_manager.name, 'Martin')
+        new_option_selector = '#id_team_manager option[value="%d"]' % new_manager.pk
+        self.assertTrue(self.selenium.find_element_by_css_selector(new_option_selector).get_attribute('selected'))
```

Comment:
@aaboffill try to install `flake8` and run it on your code.
https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/coding-style/